src/wazuh_modules/vulnerability_scanner/tests/databaseFeedManager/databaseFeedManager_test.cpp

```diff
@@ -537,3 +537,224 @@ TEST_F(DatabaseFeedManagerTest, GetVulnerabilityRemediation_InvalidData)
+{
+    // Test setup
+    const auto configurationParameters = R"( {"topicName": "topicNameTest"} )"_json;
+    std::shared_ptr<MockIndexerConnector> pIndexerConnectorMock;
+
+    spIndexerConnectorMock = std::make_shared<MockIndexerConnector>();
+
+    spPolicyManagerMock = std::make_shared<MockPolicyManager>();
+    EXPECT_CALL(*spPolicyManagerMock, getUpdaterConfiguration()).WillRepeatedly(Return(configurationParameters));
+
+    spContentRegisterMock = std::make_shared<MockContentRegister>(
+        configurationParameters.at("topicName").get<const std::string>(), configurationParameters);
+
+    spRouterSubscriberMock = std::make_shared<MockRouterSubscriber>(
+        configurationParameters.at("topicName").get<const std::string>(), "vulnerability_feed_manager");
+    EXPECT_CALL(*spRouterSubscriberMock, subscribe(_));
+
+    // Simulate saved data, store before variable instance.
+    {
+        std::string idDummy {"WT_001"};
+        flatbuffers::FlatBufferBuilder builder;
+        auto dbWrapper = std::make_unique<Utils::RocksDBWrapper>(TRANSLATIONS_DATABASE_PATH);
+        auto src = CreatesrcTrlDataDirect(builder, "^The Apache Software", "^Apache Tomcat.*", "");
+        auto tra = CreatesrcTrlDataDirect(builder, "apache", "tomcat", "");
+        std::vector<flatbuffers::Offset<srcTrlData>> translations_vector;
+        translations_vector.push_back(tra);
+        auto data = CreatedataTranslationDirect(builder, "windows", src, &translations_vector, Action_REP_PRODUCT);
+        auto dummy = CreatePackageTranslationDirect(builder, "WT_001", data);
+        builder.Finish(dummy);
+        rocksdb::Slice value(reinterpret_cast<const char*>(builder.GetBufferPointer()), builder.GetSize());
+        dbWrapper->put(idDummy, value);
+    }
+
+    auto spIndexerConnectorTramp = std::make_shared<TrampolineIndexerConnector>();
+
+    auto spDatabaseFeedManager {
+        std::make_shared<TDatabaseFeedManager<TrampolineIndexerConnector,
+                                              TrampolinePolicyManager,
+                                              TrampolineContentRegister,
+                                              TrampolineRouterSubscriber>>(spIndexerConnectorTramp)};
+    // Variables setup
+    std::string packageName {"Apache Tomcat.GenericVersion"};
+    std::vector<std::string> translations;
+
+    spDatabaseFeedManager->getWazuhPackageTranslation(
+        packageName,
+        [&translations](const NSVulnerabilityScanner::srcTrlData& translation)
+        { translations.push_back(translation.product()->str()); });
+
+    // Call function
+    spDatabaseFeedManager->getWazuhPackageTranslation(
+        packageName,
+        [&translations](const NSVulnerabilityScanner::srcTrlData& translation)
+        { translations.push_back(translation.product()->str()); });
+
+    // Assert
+    EXPECT_EQ(translations.size(), 2);
+    EXPECT_EQ(translations.at(0), "tomcat");
+    EXPECT_EQ(translations.at(1), "tomcat");
```

Comment:
The fact that the translation is found twice doesn't assure that it has got a cache hit

Reply:
According to what we talked about earlier today, yes we tested two occurrences of the package and saw the correct behavior in an indirect way, with the test output we can see:

```bash
[ RUN      ] DatabaseFeedManagerTest.PackageTranslationCacheHit
Cache miss for package 'Apache Tomcat.GenericVersion'
Cache hit for package 'Apache Tomcat.GenericVersion'
[       OK ] DatabaseFeedManagerTest.PackageTranslationCacheHit (154 ms)
```

We can change the function signature, we can check with @Dwordcito to return a code `(-1,0,1)` for missing, database and cache scenario.